ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt

```diff
@@ -320,6 +334,57 @@ private fun StackWithEdgeToEdgeBadge(
     }
 }
 
+@Suppress("LongParameterList")
+@Composable
+private fun StackWithShortEdgeToEdgeBadge(
+    stackState: StackComponentState,
+    state: PaywallState.Loaded.Components,
+    badgeStack: StackComponentStyle,
+    alignment: TwoDimensionalAlignment,
+    clickHandler: suspend (PaywallAction) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val badgeRectangleCorners = (badgeStack.shape as? Shape.Rectangle)?.corners
+        ?: CornerRadiuses(all = 20.0) // For pill badge shape default to a "medium" rounded corner.
+    val badgeShape = Shape.Rectangle(
+        corners = when (alignment) {
+            TwoDimensionalAlignment.TOP_LEADING -> CornerRadiuses(
+                topLeading = 0.0,
+                topTrailing = 0.0,
+                bottomLeading = 0.0,
+                bottomTrailing = badgeRectangleCorners.bottomTrailing,
+            )
+            TwoDimensionalAlignment.TOP_TRAILING -> CornerRadiuses(
+                topLeading = 0.0,
+                topTrailing = 0.0,
+                bottomLeading = badgeRectangleCorners.bottomLeading,
+                bottomTrailing = 0.0,
+            )
+            TwoDimensionalAlignment.BOTTOM_LEADING -> CornerRadiuses(
+                topLeading = 0.0,
+                topTrailing = badgeRectangleCorners.topTrailing,
+                bottomLeading = 0.0,
+                bottomTrailing = 0.0,
+            )
+            TwoDimensionalAlignment.BOTTOM_TRAILING -> CornerRadiuses(
+                topLeading = badgeRectangleCorners.topLeading,
+                topTrailing = 0.0,
+                bottomLeading = 0.0,
+                bottomTrailing = 0.0,
+            )
+            else -> CornerRadiuses(all = 0.0)
+        },
+    )
+    MainStackComponent(stackState, state, clickHandler, modifier) {
+        StackComponentView(
```

Comment:
Currently, the badge can be clipped by the corners of the main stack... I think that developers can fix this by adding a padding in the dashboard so it doesn't overlap. We could potentially add some padding manually ourselves, but I think it's a lot of extra logic, and probably less flexible.

This can be a bit of a problem with pill shaped main stacks since the corner is dynamic... But usually an approximate padding should work. 

We can always revisit if needed